src/lib/sidenav/sidenav.ts

```diff
@@ -161,7 +162,7 @@ export class MdSidenav implements AfterContentInit, OnDestroy {
     let activeEl = this._doc && this._doc.activeElement;
     if (activeEl && this._elementRef.nativeElement.contains(activeEl)) {
       if (this._elementFocusedBeforeSidenavWasOpened instanceof HTMLElement) {
-        this._elementFocusedBeforeSidenavWasOpened.focus();
+        this._focusOriginMonitor.focusVia(this._elementFocusedBeforeSidenavWasOpened, 'program');
```

Comment:
This is going to do the same thing that just calling `.focus()` was doing. I think we want logic like this:
1. If the sidenav close is triggered by keyboard interaction (e.g. pressing escape) set it to `keyboard` focused
2. If the sidenav close is triggered by mouse interaction (e.g. clicking on backdrop) set it to `mouse` focused
3. If the sidenav close is triggered by directly calling `close` or `toggle` method allow the user to optionally supply the focus origin and default to `program` if they don't specify.

I think opening the sidenav needs similar logic but it's a little trickier because the sidenav isn't directly responsible for setting the focus on open, it goes through the focus trap.

You can test if this is working properly by opening and closing the sidenav in the demo app via different means and seeing what the resulting focus effect looks like